### PR TITLE
Fix GitHub action error and improve readability and efficiency

### LINKED ISSUE #160: Docker push is failing
release-docker-image / release-docker-image
Failed in 3 seconds

invalid argument "cord:/develop" for "-t, --tag" flag: invalid reference format
See 'docker build --help'.
Error: Process completed with exit code 125.

### PR DESCRIPTION
**Pull request by roynrishingha**
This commit fixes the error encountered in the GitHub action by adding a missing step to the workflow. Additionally, the code has been refactored to improve readability and efficiency. The branch name is now retrieved using a simpler and more efficient approach, and the Docker image is now tagged with the repository name and commit hash in a more concise way.

This PR fixes #160

### CODE REVIEW

**Comment by smohan-dw** [comment]
Does this integrate distro-less builds?

**Review by amarts — APPROVED** [review]

**Comment by amarts** [comment]
Distroless is done in https://github.com/dhiway/cord/blob/develop/Dockerfile which is used in these release CI jobs. So, yes, the image from this would be distroless.

File: .github/workflows/release.yml
Comment (Rithwik0480) [inline .github/workflows/release.yml:1]: The docker image was successfully pushed.

**Review by Rithwik0480 — APPROVED** [review]

**Review by smohan-dw — APPROVED** [review]

**Comment by roynrishingha** [comment]
We have branch protection rules on cord repo. Only @smohan-dw can merge on `develop` branch, so that no one accidentally push into develop. The expected workflow is to raise a PR.

**Pull request merged**